### PR TITLE
adding tests for `state_creator` of `import_object`

### PR DESCRIPTION
**Pull request by YaronWittenstein**
Part of the PR #807 changes was adding support for shared import objects between threads.

https://github.com/wasmerio/wasmer/pull/807/files#diff-d20cb4c5a883566b85be4cc046f45aa9R49

I've added tests/examples on how to create an `import object` with a state_creator
(function or closure)

### CODE REVIEW

File: lib/runtime-core/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#![deny(
+#![allow(
```
Comment (MarkMcCaskey) [inline lib/runtime-core/src/lib.rs]: This should be changed back!

File: lib/runtime-core/src/import.rs [hunk 2]
```diff
@@ -254,6 +254,56 @@ mod test {
...
+
+        let (state, _dtor) = imports.call_state_creator().unwrap();
+        let data: &mut Data = unsafe { &mut *(state as *mut Data) };
+
+        assert_eq!(ptr, data.inner);
+    }
+
+    #[test]
+    fn state_creator_closure() {
+        let ptr = std::ptr::null();
```
Comment (MarkMcCaskey) [inline lib/runtime-core/src/import.rs]: Perhaps we should use a sentinel value, `0` is a common bit pattern in memory, something like:

`let ptr = unsafe { std::mem::transmute(0xB7B7A0A01234) };`

would make the test more robust I think

File: lib/runtime-core/src/lib.rs (hunk 1 above)
Comment (YaronWittenstein) [reply lib/runtime-core/src/lib.rs]: @MarkMcCaskey  may bad

**Comment by YaronWittenstein** [comment]
> bors r+

@MarkMcCaskey can you please re-trigger `bors` ?

**Comment by MarkMcCaskey** [comment]
Thanks!  Sorry about not shipping this before -- I only very recently learned/realized that not everyone can use bors on Wasmer! I assumed given an approval, people could ship their own PRs, but this seems to not be the case!  There are a few other PRs I didn't ship due to this misunderstanding which I'll be following up on

**Comment by syrusakbary** [comment]
This is no longer needed on the refactor. Adding the label so we can close quickly the PR after publishing the refactor.

**Comment by syrusakbary** [comment]
State creator is no longer supported in the import object.
We moved towards a cleaner API in the refactor, removing the need for state creation.

**Pull request closed without merging**